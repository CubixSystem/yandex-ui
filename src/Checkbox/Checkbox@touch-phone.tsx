import { withRegistry } from '@bem-react/di';

import { Checkbox as CheckboxCommon } from './Checkbox';
import { checkboxRegistry } from './Checkbox.registry/touch-pad';

export * from './Checkbox';

export const Checkbox = withRegistry(checkboxRegistry)(CheckboxCommon);
