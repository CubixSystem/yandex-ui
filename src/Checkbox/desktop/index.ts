export * from '../Checkbox@desktop';
export * from '../_indeterminate/Checkbox_indeterminate';
export * from '../_lines/Checkbox_lines_multi';
export * from '../_lines/Checkbox_lines_one';
export * from '../_size/Checkbox_size_m';
export * from '../_size/Checkbox_size_s';
export * from '../_theme/Checkbox_theme_normal@desktop';
export * from '../_theme/Checkbox_theme_pseudo@desktop';
export * from '../_view/Checkbox_view_default@desktop';
export * from '../_view/Checkbox_view_outline@desktop';
