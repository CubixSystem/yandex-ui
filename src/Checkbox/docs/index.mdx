---
id: checkbox
path: /components/checkbox

title: Checkbox
description: Компонент для создания чекбоксов различных типов.

links:
 - label: source
   url: https://github.yandex-team.ru/search-interfaces/frontend/tree/master/packages/lego-components/src/Checkbox
 - label: startrek
   url: https://search.yandex-team.ru/stsearch?text=Checkbox.ts&facet.queue=ISL&facet.type=bug&facet.status=128
 - label: figma
   url: https://www.figma.com/file/CaoHFEO0w6PaE8cVaYFDoq8i/L-E-G-O?node-id=0%3A1

tabs:
 - Examples
 - Usage
 - Playground
 - Props
 - Accessibility
---
