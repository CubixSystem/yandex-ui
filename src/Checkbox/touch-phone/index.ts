export * from '../Checkbox@touch-phone';
export * from '../_indeterminate/Checkbox_indeterminate';
export * from '../_lines/Checkbox_lines_multi';
export * from '../_lines/Checkbox_lines_one';
export * from '../_size/Checkbox_size_m';
export * from '../_size/Checkbox_size_s';
export * from '../_theme/Checkbox_theme_normal';
export * from '../_theme/Checkbox_theme_pseudo';
export * from '../_view/Checkbox_view_default';
export * from '../_view/Checkbox_view_outline';
