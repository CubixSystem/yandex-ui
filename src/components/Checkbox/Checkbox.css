/**
 * 1. Нужно, чтобы Label всегда оставался на одной строке с Box.
 */

.Checkbox {
    font-family: var(--control-font-family);
    user-select: none;
    white-space: nowrap; /* 1 */

    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);

    touch-action: manipulation;
}
