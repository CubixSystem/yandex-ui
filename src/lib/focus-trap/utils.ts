import { getTabbables } from './tabbable';

const FOCUS_TRAP_GUARD_ATTR = 'focusTrapGuard';

interface ScrollableElement {
    element: HTMLElement;
    scrollTop: number;
    scrollLeft: number;
}

let supportsPreventScrollCached: boolean | null = null;
function supportsPreventScroll() {
    if (supportsPreventScrollCached === null) {
        supportsPreventScrollCached = false;

        try {
            const element = document.createElement('div');

            element.focus({
                get preventScroll() {
                    supportsPreventScrollCached = true;
                    return true;
                },
            });
        } catch (e) {
            // ignore
        }
    }

    return supportsPreventScrollCached;
}

function getScrollableElements(element: HTMLElement): ScrollableElement[] {
    const scrollableElements: ScrollableElement[] = [];
    const rootScrollingElement = document.scrollingElement || document.documentElement;

    let parent = element.parentNode;
    while (parent instanceof HTMLElement && parent !== rootScrollingElement) {
        if (parent.offsetHeight < parent.scrollHeight || parent.offsetWidth < parent.scrollWidth) {
            scrollableElements.push({
                element: parent,
                scrollTop: parent.scrollTop,
                scrollLeft: parent.scrollLeft,
            });
        }
        parent = parent.parentNode;
    }

    if (rootScrollingElement instanceof HTMLElement) {
        scrollableElements.push({
            element: rootScrollingElement,
            scrollTop: rootScrollingElement.scrollTop,
            scrollLeft: rootScrollingElement.scrollLeft,
        });
    }

    return scrollableElements;
}

function restoreScrollPosition(scrollableElements: ScrollableElement[]) {
    for (const { element, scrollTop, scrollLeft } of scrollableElements) {
        element.scrollTop = scrollTop;
        element.scrollLeft = scrollLeft;
    }
}

export function getActiveElement() {
    return document.activeElement as HTMLElement | null;
}

export function isElementInScope(scope: Element, element: Element) {
    return scope.contains(element);
}

export function isFocusFree() {
    return getActiveElement() === document.body;
}

export function createGuard(tabIndex: number) {
    const element = document.createElement('span');
    element.tabIndex = tabIndex;
    Object.assign(element.style, {
        width: '1px',
        height: 0,
        padding: 0,
        overflow: 'hidden',
        position: 'fixed',
        top: '1px',
        left: '1px',
    });
    element.setAttribute('aria-hidden', 'true');
    element.dataset[FOCUS_TRAP_GUARD_ATTR] = 'true';

    return element;
}

export function isGuard(node: HTMLElement) {
    return Boolean(node.dataset[FOCUS_TRAP_GUARD_ATTR]);
}

export function removeElementFromDocument(element: Element) {
    if (!element.parentNode) {
        return;
    }

    element.parentNode.removeChild(element);
}

function focusWithoutScrolling(element: HTMLElement) {
    if (supportsPreventScroll()) {
        element.focus({ preventScroll: true });
    } else {
        const scrollableElements = getScrollableElements(element);
        element.focus();
        restoreScrollPosition(scrollableElements);
    }
}

export function focusElement(element?: HTMLElement | null, preventScroll?: boolean) {
    if (!element) {
        return;
    }

    try {
        if (preventScroll) {
            focusWithoutScrolling(element);
        } else {
            element.focus();
        }
    } catch (e) {
        // ignore
    }
}

export function focusFirstIn(scope: HTMLElement) {
    const [first] = getTabbables(scope, (node) => !isGuard(node));

    focusElement(first);
}

export function getNextTabbableIndex(current: number, count: number, step: number) {
    if (current === -1) {
        return 0;
    }

    return count > 0 ? (count + current + step) % count : -1;
}
