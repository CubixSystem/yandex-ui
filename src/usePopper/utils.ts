import { Placement, BasePlacement, Offsets, top, left, right } from '@popperjs/core';

export function getLayoutRect(element: HTMLElement) {
    return {
        x: element.offsetLeft,
        y: element.offsetTop,
        width: element.offsetWidth,
        height: element.offsetHeight,
    };
}

export default function getBasePlacement(placement: Placement): BasePlacement {
    return placement.split('-')[0] as any;
}

export function distanceAndSkiddingToXY(placement: Placement, offset: [number, number]): Offsets {
    const basePlacement = getBasePlacement(placement);
    const invertDistance = [left, top].indexOf(basePlacement as any) >= 0 ? -1 : 1;

    const skidding = offset[0] || 0;
    const distance = (offset[1] || 0) * invertDistance;

    return [left, right].indexOf(basePlacement as any) >= 0
        ? { x: distance, y: skidding }
        : { x: skidding, y: distance };
}
