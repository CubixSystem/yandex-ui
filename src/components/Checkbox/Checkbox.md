# Checkbox

<a href='https://github.yandex-team.ru/search-interfaces/frontend/tree/master/packages/lego-components/src/components/Checkbox' target='_blank'><img src='https://badger.yandex-team.ru/custom/[Исходники]/[Github
][green]/badge.svg' /></a> <a href='https://search.yandex-team.ru/stsearch?text=Checkbox.ts&facet.queue=ISL&facet.type=bug&facet.status=128' target='_blank'><img src='https://badger.yandex-team.ru/custom/[Известные проблемы]/[Startrek][blue]/badge.svg' /></a>

<!-- description:start -->
Компонент для создания чекбоксов различных типов.
<!-- description:end -->

## Пример использования

Конфигурация темы на уровне проекта:

```ts
// src/lib/theme.ts
import { configureRootTheme } from '@yandex-lego/components/Theme'
import { theme } from '@yandex-lego/components/Theme/presets/default'

configureRootTheme({ theme })
```

Использование с нужным набором модификаторов:

```ts
// src/App.ts
import React, { useState } from 'react'
import { compose } from '@bem-react/core'
import {
  Checkbox as CheckboxDesktop,
  withSizeM,
  withViewDefault,
} from '@yandex-lego/components/Checkbox/desktop'

// Композиция из различных модификаторов
const Checkbox = compose(withSizeM, withViewDefault)(CheckboxDesktop)

const App = () => {
  const [checked, setChecked] = useState(false)

  return (
    <Checkbox
      label="checkbox"
      size="m"
      view="default"
      onChange={() => setChecked(!checked)}
      checked={checked}
    />
  )
}
```

Использование с полным набором модификаторов:

```ts
// src/App.ts
import React, { useState } from 'react'
import { Checkbox } from '@yandex-lego/components/Checkbox/desktop/bundle'

const App = () => {
  const [checked, setChecked] = useState(false)

  return (
    <Checkbox
      label="checkbox"
      size="m"
      view="default"
      onChange={() => setChecked(!checked)}
      checked={checked}
    />
  )
}
```

## Примеры

### Вид чекбокса

Чтобы изменить вид чекбокса, установите свойство `view` в значение `dafault`.

{{%story::desktop:controls-checkbox-desktop--view%}}

### Стилевое оформление

Чтобы изменить стилевое оформление чекбокса, установите свойство `theme` в одно из следующих значений:

- `normal` — стандартная тема. Используется в большинстве случаев.
- `pseudo` — прозрачная тема. Используется, чтобы сделать переключатель менее заметным на странице.

{{%story::desktop:controls-checkbox-desktop--theme%}}

### Размер

Чтобы изменить размер чекбокса, установите свойство `size` в одно из следующих значений: `"m"`, `"s"`.

{{%story::desktop:controls-checkbox-desktop--size%}}

### Однострочные и многострочные подписи

Чтобы задать тип подписи чекбоксу, установите свойство `lines` в одно из следующих значений:

- `one` — Однострочная подпись. Обрезается троеточием, если ее длина превышает длину родительского элемента.
- `multi` — Многострочная подпись.

{{%story::desktop:controls-checkbox-desktop--lines%}}

### Переходное состояние

Чтобы задать переходное состояние чекбоксу, установите свойство `indeterminate` в значение `true`.

{{%story::desktop:controls-checkbox-desktop--indeterminate%}}

## Свойства

<!-- props:start -->
| Свойство       | Тип                                                         | Описание                                                                                                                                                                                                                                                                                                                                                                            |
| -------------- | ----------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| onBlur?        | `(event: FocusEvent<HTMLElement>) => void`                  | Событие, которое вызывается при потере фокуса компонентом. Например, при клике на другом месте экрана                                                                                                                                                                                                                                                                               |
| onClick?       | `(event: MouseEvent<HTMLElement, MouseEvent>) => void`      | Событие, которое вызывается при нажатии на компонент                                                                                                                                                                                                                                                                                                                                |
| onFocus?       | `(event: FocusEvent<HTMLElement>) => void`                  | Событие, которое возникает при получении компонентом фокуса                                                                                                                                                                                                                                                                                                                         |
| onMouseDown?   | `(event: MouseEvent<HTMLElement, MouseEvent>) => void`      | Событие по своему действию похоже на `onClick` и возникает в момент нажатия кнопки мыши. `onClick` в каком-то смысле является комбинацией событий `onMouseDown` и `onMouseUp`                                                                                                                                                                                                       |
| onMouseUp?     | `(event: MouseEvent<HTMLElement, MouseEvent>) => void`      | Событие по своему действию противоположно событию `onMouseDown` и происходит при отпускании кнопки мыши. Курсор должен находится в пределах HTML-элемента, к которому добавлен атрибут `onmouseup`                                                                                                                                                                                  |
| onMouseLeave?  | `(event: MouseEvent<HTMLElement, MouseEvent>) => void`      | Обработчик события `onMouseLeave`                                                                                                                                                                                                                                                                                                                                                   |
| onMouseEnter?  | `(event: MouseEvent<HTMLElement, MouseEvent>) => void`      | Обработчик события `onMouseEnter`                                                                                                                                                                                                                                                                                                                                                   |
| focused?       | `false \| true`                                             | Состояние фокуса на компоненте                                                                                                                                                                                                                                                                                                                                                      |
| disabled?      | `false \| true`                                             | Неактивное состояние компонента. Состояние, при котором компонент отображается, но недоступен для действий пользователя                                                                                                                                                                                                                                                             |
| pressed?       | `false \| true`                                             | Состояние нажатия на компоненте                                                                                                                                                                                                                                                                                                                                                     |
| hovered?       | `false \| true`                                             | Состояние, которое возникает при наведении на компонент курсором                                                                                                                                                                                                                                                                                                                    |
| innerRef?      | `(instance: HTMLElement) => void \| RefObject<HTMLElement>` | Ссылка на корневой DOM-элемент компонента                                                                                                                                                                                                                                                                                                                                           |
| label?         | `string`                                                    | Текст подписи к чекбоксу                                                                                                                                                                                                                                                                                                                                                            |
| indeterminate? | `false \| true`                                             | Визуально переводит чекбокс в неопределенное состояние. Не влияет на состояние, указанное в `checked`.<br>Может использоваться в дереве чекбоксов, чтобы показать состояние родительского чекбокса, когда хотя бы один вложенный чекбокс отмечен, но не все.<br>Если свойство задано родительскому чекбоксу, то в `aria-controls` необходимо добавить `id` всех вложенных чекбоксов |
| checked?       | `false \| true`                                             | Состояние переключателя: включен или выключен                                                                                                                                                                                                                                                                                                                                       |
| controlRef?    | `(instance: T) => void \| RefObject<T>`                     | Ссылка на дом-ноду нативного контрола.                                                                                                                                                                                                                                                                                                                                              |
| className?     | `string`                                                    | Дополнительный класс                                                                                                                                                                                                                                                                                                                                                                |
| onChange?      | `(event: ChangeEvent<HTMLInputElement>) => void`            | Обработчик изменения значения                                                                                                                                                                                                                                                                                                                                                       |
| name?          | `string`                                                    | Имя компонента                                                                                                                                                                                                                                                                                                                                                                      |
| autoComplete?  | `string`                                                    | HTML-атрибут `autoComplete`                                                                                                                                                                                                                                                                                                                                                         |
| id?            | `string`                                                    | Уникальный id компонента                                                                                                                                                                                                                                                                                                                                                            |
| onKeyUp?       | `(event: KeyboardEvent<HTMLInputElement>) => void`          | Обработчик события `onKeyUp`                                                                                                                                                                                                                                                                                                                                                        |
| onKeyDown?     | `(event: KeyboardEvent<HTMLInputElement>) => void`          | Обработчик события `onKeyDown`                                                                                                                                                                                                                                                                                                                                                      |
<!-- props:end -->

## a11y

При использовании свойства `indeterminate` для обеспечения доступности необходимо добавить в свойство `aria-controls` все `id` дочерних элементов.

## Смотрите также

- <a href='https://www.w3.org/TR/wai-aria-practices/#checkbox' target='_blank'>WAI-ARIA Authoring Practices</a>
