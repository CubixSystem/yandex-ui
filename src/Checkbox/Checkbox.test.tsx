import React, { createRef, ComponentType } from 'react';
import { mount, render } from 'enzyme';
import { ExtractProps } from '@bem-react/core';

import { Checkbox as CheckboxCommon } from './Checkbox';
import { Checkbox as CheckboxDesktop } from './Checkbox@desktop';
import { Checkbox as CheckboxTouchPad } from './Checkbox@touch-pad';
import { Checkbox as CheckboxTouchPhone } from './Checkbox@touch-phone';
import { serverEnvironmentSetup, delay } from '../internal/utils';

const platforms = [['desktop', CheckboxDesktop], ['touch-pad', CheckboxTouchPad], ['touch-phone', CheckboxTouchPhone]];

type CheckboxProps = ExtractProps<typeof CheckboxDesktop>;

describe.each<any>(platforms)('Checkbox@%s', (_platform, Checkbox: ComponentType<CheckboxProps>) => {
    describe('server environment', () => {
        serverEnvironmentSetup();

        test('должен вернуть полный шаблон компонента (snapshot)', () => {
            expect(render(<Checkbox id="serverTemplateId" className="mix" view="default" />)).toMatchSnapshot();
        });
    });

    describe('client environment', () => {
        test('должен вернуть полный шаблон компонента (snapshot)', () => {
            expect(render(<Checkbox id="clientTemplateId" className="mix" view="default" />)).toMatchSnapshot();
        });

        test('должен проставить в label произвольный tsx (snapshot)', () => {
            expect(render(<Checkbox label={<div className="custom">content</div>} view="default" />)).toMatchSnapshot();
        });

        test('должен быть установлен корректный displayName', () => {
            //Так написано, потому что компонент сразу поставляется в обертке.
            expect(CheckboxCommon.displayName).toBe('WithControl(Checkbox)');
        });

        test('должен устанавливать ссылку на корневой DOM элемент', async () => {
            const innerRef = createRef<HTMLElement>();
            mount(<Checkbox innerRef={innerRef} />);
            await delay(100);
            expect(innerRef.current).not.toBe(null);
        });

        test('должен устанавливать ссылку на нативный DOM элемент', async () => {
            const controlRef = createRef<HTMLInputElement>();
            mount(<Checkbox controlRef={controlRef} />);
            await delay(100);
            expect(controlRef.current).not.toBe(null);
        });

        test('должен вызывать событие onChange', () => {
            const onChange = jest.fn();
            const wrapper = mount(<Checkbox onChange={onChange} />);
            wrapper.find('.Checkbox-Control').simulate('change');
            expect(onChange).toHaveBeenCalledTimes(1);
        });

        test('должен рендерить label для checkbox', () => {
            const wrapper = mount(<Checkbox label="label" />);
            expect(wrapper.find('.Checkbox-Label')).toHaveLength(1);
        });

        test('должен устанавливать aria-checked/checked при обновлении свойства checked/indeterminate', () => {
            const wrapper = mount(<Checkbox checked={false} onChange={() => null} />);

            wrapper.setProps({ checked: true });
            expect(wrapper.find('.Checkbox-Control').prop('aria-checked')).toBe(true);
            expect(wrapper.find('.Checkbox-Control').prop('checked')).toBe(true);

            wrapper.setProps({ checked: false });
            expect(wrapper.find('.Checkbox-Control').prop('aria-checked')).toBe(false);
            expect(wrapper.find('.Checkbox-Control').prop('checked')).toBe(false);

            wrapper.setProps({ indeterminate: true });
            expect(wrapper.find('.Checkbox-Control').prop('aria-checked')).toBe('mixed');
            expect(wrapper.find('.Checkbox-Control').prop('checked')).toBe(true);

            wrapper.setProps({ indeterminate: false });
            expect(wrapper.find('.Checkbox-Control').prop('aria-checked')).toBe(false);
            expect(wrapper.find('.Checkbox-Control').prop('checked')).toBe(false);
        });

        test('должен устанавливать свойство indeterminate у control при обновлении свойства indeterminate', async () => {
            const controlRef = createRef<HTMLInputElement>();
            const wrapper = mount(<Checkbox checked={false} controlRef={controlRef} onChange={() => null} />);
            await delay(100);

            wrapper.setProps({ indeterminate: true });
            expect(controlRef.current).toHaveProperty('indeterminate', true);
            wrapper.setProps({ indeterminate: false });
            expect(controlRef.current).toHaveProperty('indeterminate', false);
        });

        test('должен возвращать aria-checked/checked после обновления indeterminate если свойство checked установлено', () => {
            const wrapper = mount(<Checkbox checked={false} onChange={() => null} />);

            wrapper.setProps({ checked: true });
            expect(wrapper.find('.Checkbox-Control').prop('aria-checked')).toBe(true);

            wrapper.setProps({ indeterminate: true });
            expect(wrapper.find('.Checkbox-Control').prop('aria-checked')).toBe('mixed');

            wrapper.setProps({ indeterminate: false });
            expect(wrapper.find('.Checkbox-Control').prop('aria-checked')).toBe(true);
        });
    });
});
