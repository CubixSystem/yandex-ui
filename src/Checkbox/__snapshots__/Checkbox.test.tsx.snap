// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox@desktop client environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Checkbox mix"
>
  <span
    class="Checkbox-Box"
  >
    <input
      aria-labelledby="label-clientTemplateId"
      autocomplete="off"
      class="Checkbox-Control"
      id="clientTemplateId"
      type="checkbox"
    />
    <span
      class="Checkbox-Tick"
    >
      <span
        aria-hidden="true"
        class="Icon Icon_hasGlyph Icon_glyph_type-tick"
      >
        <svg
          height="10"
          viewBox="0 0 14 10"
          width="14"
        >
          <path
            d="M1.49 4.885l1.644-1.644 4.385 4.385L5.874 9.27 1.49 4.885zm4.384 1.096L11.356.5 13 2.144 7.519 7.626 5.874 5.98v.001z"
          />
        </svg>
      </span>
    </span>
  </span>
</span>
`;

exports[`Checkbox@desktop client environment должен проставить в label произвольный tsx (snapshot) 1`] = `
<span
  class="Checkbox"
>
  <span
    class="Checkbox-Box"
  >
    <input
      aria-labelledby="label-xuniq-0-1"
      autocomplete="off"
      class="Checkbox-Control"
      id="xuniq-0-1"
      type="checkbox"
    />
    <span
      class="Checkbox-Tick"
    >
      <span
        aria-hidden="true"
        class="Icon Icon_hasGlyph Icon_glyph_type-tick"
      >
        <svg
          height="10"
          viewBox="0 0 14 10"
          width="14"
        >
          <path
            d="M1.49 4.885l1.644-1.644 4.385 4.385L5.874 9.27 1.49 4.885zm4.384 1.096L11.356.5 13 2.144 7.519 7.626 5.874 5.98v.001z"
          />
        </svg>
      </span>
    </span>
  </span>
  <label
    aria-hidden="true"
    class="Checkbox-Label"
    for="xuniq-0-1"
    id="label-xuniq-0-1"
  >
    <div
      class="custom"
    >
      content
    </div>
  </label>
</span>
`;

exports[`Checkbox@desktop server environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Checkbox mix"
>
  <span
    class="Checkbox-Box"
  >
    <input
      aria-labelledby="label-serverTemplateId"
      autocomplete="off"
      class="Checkbox-Control"
      id="serverTemplateId"
      type="checkbox"
    />
    <span
      class="Checkbox-Tick"
    >
      <span
        aria-hidden="true"
        class="Icon Icon_hasGlyph Icon_glyph_type-tick"
      >
        <svg
          height="10"
          viewBox="0 0 14 10"
          width="14"
        >
          <path
            d="M1.49 4.885l1.644-1.644 4.385 4.385L5.874 9.27 1.49 4.885zm4.384 1.096L11.356.5 13 2.144 7.519 7.626 5.874 5.98v.001z"
          />
        </svg>
      </span>
    </span>
  </span>
</span>
`;

exports[`Checkbox@touch-pad client environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Checkbox mix"
>
  <span
    class="Checkbox-Box"
  >
    <input
      aria-labelledby="label-clientTemplateId"
      autocomplete="off"
      class="Checkbox-Control"
      id="clientTemplateId"
      type="checkbox"
    />
    <span
      class="Checkbox-Tick"
    >
      <span
        aria-hidden="true"
        class="Icon Icon_hasGlyph Icon_glyph_type-tick"
      >
        <svg
          height="10"
          viewBox="0 0 14 10"
          width="14"
        >
          <path
            d="M1.49 4.885l1.644-1.644 4.385 4.385L5.874 9.27 1.49 4.885zm4.384 1.096L11.356.5 13 2.144 7.519 7.626 5.874 5.98v.001z"
          />
        </svg>
      </span>
    </span>
  </span>
</span>
`;

exports[`Checkbox@touch-pad client environment должен проставить в label произвольный tsx (snapshot) 1`] = `
<span
  class="Checkbox"
>
  <span
    class="Checkbox-Box"
  >
    <input
      aria-labelledby="label-xuniq-0-9"
      autocomplete="off"
      class="Checkbox-Control"
      id="xuniq-0-9"
      type="checkbox"
    />
    <span
      class="Checkbox-Tick"
    >
      <span
        aria-hidden="true"
        class="Icon Icon_hasGlyph Icon_glyph_type-tick"
      >
        <svg
          height="10"
          viewBox="0 0 14 10"
          width="14"
        >
          <path
            d="M1.49 4.885l1.644-1.644 4.385 4.385L5.874 9.27 1.49 4.885zm4.384 1.096L11.356.5 13 2.144 7.519 7.626 5.874 5.98v.001z"
          />
        </svg>
      </span>
    </span>
  </span>
  <label
    aria-hidden="true"
    class="Checkbox-Label"
    for="xuniq-0-9"
    id="label-xuniq-0-9"
  >
    <div
      class="custom"
    >
      content
    </div>
  </label>
</span>
`;

exports[`Checkbox@touch-pad server environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Checkbox mix"
>
  <span
    class="Checkbox-Box"
  >
    <input
      aria-labelledby="label-serverTemplateId"
      autocomplete="off"
      class="Checkbox-Control"
      id="serverTemplateId"
      type="checkbox"
    />
    <span
      class="Checkbox-Tick"
    >
      <span
        aria-hidden="true"
        class="Icon Icon_hasGlyph Icon_glyph_type-tick"
      >
        <svg
          height="10"
          viewBox="0 0 14 10"
          width="14"
        >
          <path
            d="M1.49 4.885l1.644-1.644 4.385 4.385L5.874 9.27 1.49 4.885zm4.384 1.096L11.356.5 13 2.144 7.519 7.626 5.874 5.98v.001z"
          />
        </svg>
      </span>
    </span>
  </span>
</span>
`;

exports[`Checkbox@touch-phone client environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Checkbox mix"
>
  <span
    class="Checkbox-Box"
  >
    <input
      aria-labelledby="label-clientTemplateId"
      autocomplete="off"
      class="Checkbox-Control"
      id="clientTemplateId"
      type="checkbox"
    />
    <span
      class="Checkbox-Tick"
    >
      <span
        aria-hidden="true"
        class="Icon Icon_hasGlyph Icon_glyph_type-tick"
      >
        <svg
          height="10"
          viewBox="0 0 14 10"
          width="14"
        >
          <path
            d="M1.49 4.885l1.644-1.644 4.385 4.385L5.874 9.27 1.49 4.885zm4.384 1.096L11.356.5 13 2.144 7.519 7.626 5.874 5.98v.001z"
          />
        </svg>
      </span>
    </span>
  </span>
</span>
`;

exports[`Checkbox@touch-phone client environment должен проставить в label произвольный tsx (snapshot) 1`] = `
<span
  class="Checkbox"
>
  <span
    class="Checkbox-Box"
  >
    <input
      aria-labelledby="label-xuniq-0-17"
      autocomplete="off"
      class="Checkbox-Control"
      id="xuniq-0-17"
      type="checkbox"
    />
    <span
      class="Checkbox-Tick"
    >
      <span
        aria-hidden="true"
        class="Icon Icon_hasGlyph Icon_glyph_type-tick"
      >
        <svg
          height="10"
          viewBox="0 0 14 10"
          width="14"
        >
          <path
            d="M1.49 4.885l1.644-1.644 4.385 4.385L5.874 9.27 1.49 4.885zm4.384 1.096L11.356.5 13 2.144 7.519 7.626 5.874 5.98v.001z"
          />
        </svg>
      </span>
    </span>
  </span>
  <label
    aria-hidden="true"
    class="Checkbox-Label"
    for="xuniq-0-17"
    id="label-xuniq-0-17"
  >
    <div
      class="custom"
    >
      content
    </div>
  </label>
</span>
`;

exports[`Checkbox@touch-phone server environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Checkbox mix"
>
  <span
    class="Checkbox-Box"
  >
    <input
      aria-labelledby="label-serverTemplateId"
      autocomplete="off"
      class="Checkbox-Control"
      id="serverTemplateId"
      type="checkbox"
    />
    <span
      class="Checkbox-Tick"
    >
      <span
        aria-hidden="true"
        class="Icon Icon_hasGlyph Icon_glyph_type-tick"
      >
        <svg
          height="10"
          viewBox="0 0 14 10"
          width="14"
        >
          <path
            d="M1.49 4.885l1.644-1.644 4.385 4.385L5.874 9.27 1.49 4.885zm4.384 1.096L11.356.5 13 2.144 7.519 7.626 5.874 5.98v.001z"
          />
        </svg>
      </span>
    </span>
  </span>
</span>
`;
